fmt/src/visit.rs

```diff
@@ -3,37 +3,42 @@
 use crate::solang_ext::*;
 use solang_parser::pt::*;
 
-/// The error type a [Visitor] may return
-pub type VError = Box<dyn std::error::Error>;
-/// The result type a [Visitor] may return
-pub type VResult = Result<(), VError>;
-
 /// A trait that is invoked while traversing the Solidity Parse Tree.
 /// Each method of the [Visitor] trait is a hook that can be potentially overridden.
 ///
 /// Currently the main implementor of this trait is the [`Formatter`](crate::Formatter) struct.
 pub trait Visitor {
-    fn visit_source(&mut self, _loc: Loc) -> VResult {
+    type Error: std::error::Error;
```

Comment:
this should be fine but defer to @rkrasiuk here whether this will cause some problems with docs etc.

since we're the only consumer of this trait perhaps it's fine to just change the `VResult` type